Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -474,16 +506,16 @@ protected override void ProcessRecord()
                 catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Forbidden)
                 {
                     WriteError(new ErrorRecord(ex, 
-                        $"You do not have permission to delete {InputObject.DatasetReference.DatasetId}.",
+                        $"You do not have permission to delete {Dataset}.",
                         ErrorCategory.PermissionDenied, 
-                        InputObject));
+                        Dataset));
                 } 
             }
         }
 
-        private string GetConfirmationMessage(Dataset d, int? tables)
+        private string GetConfirmationMessage(string datasetId, int? tables)
         {
-            return $@"{d.DatasetReference.DatasetId} has {tables} tables and the -Force parameter was not specified. "
+            return $@"{datasetId} has {tables} tables and the -Force parameter was not specified. "
```

Comment:
You should add single quote (') around {datasetId} and {tables} to make it more readable.